src/ui/common/src/components/workflows/ReactFlowCanvas.tsx

```diff
@@ -27,11 +27,30 @@ const ReactFlowCanvas: React.FC<ReactFlowCanvasProps> = ({
 
   const { edges, nodes } = dagPositionState.result ?? { edges: [], nodes: [] };
 
+  const canvasEdges = edges.map((edge) => {
+    return {
+      id: edge.id,
+      source: edge.source,
+      target: edge.target,
+      type: edge.type,
+      container: 'root',
+    };
+  });
+
+  const canvasNodes = nodes.map((node) => {
```

Comment:
Having some extra keys in our input nodes and edges caused some issues in how the nodes were positioned. This function simply maps our nodes to ensure that the node and edge objects have the proper shape.